lib/siwapp/invoices/item.ex

```diff
@@ -24,7 +29,30 @@ defmodule Siwapp.Invoices.Item do
     |> cast_assoc(:taxes)
     |> foreign_key_constraint(:invoice_id)
     |> validate_length(:description, max: 20000)
-    |> validate_number(:quantity, greater_than_or_equal_to: 0)
-    |> validate_number(:discount, greater_than_or_equal_to: 0)
+    |> validate_number(:quantity, greater_than_or_equal_to: 1)
+    |> validate_number(:discount, greater_than_or_equal_to: 0, less_than_or_equal_to: 100)
+  end
+
+  def set_virtual_fields(item) do
+    item
+    |> Map.put(:net_amount, net_amount(item))
```

Comment:
you have to do this in the changeset and use put_change method. Talk with @RodrigoMolinaMadrid he's done something very similar